ethportal-api/src/types/execution/transaction.rs

```diff
@@ -213,6 +169,96 @@ impl<'de> Deserialize<'de> for Transaction {
     }
 }
 
+/// The wrapper around [Transaction] structure, that has different RLP encoding.
+///
+/// In most cases, RLP encoding of [Transaction] object is the correct one, but in some rare cases,
+/// this one is used instead. If unsure, use basic type.
+///
+/// The RLP encoding of this type is done in a following way:
+///
+/// - Legacy transactions are always encoded in the same way: `rlp(tx)`.
+/// - Other transactions are encoded as: `rlp(type + rlp(tx))` (where `+` represents concatenation)
+///   - `type` is a single byte represending the transaction type
+///
+/// The basic type differs in the way it encodes non-legacy transaction. The basic type doesn't have
+/// extra RLP wrapper around it, meaning it's just: `type + rlp(tx)`.
+pub struct TransactionWithRlpHeader(pub Transaction);
```

Comment:
Great doc

I understand the name `TransactionWithRlpHeader`. I don't 100% love it, though I am not sure I've got a better one. I'll talk through my thought process for why, and offer some alternatives. Feel free to ignore it all, no response necessary. :)

For a first-time code reader to quickly grok the name, it sort of assumes that the reader is actively aware of the internal steps of RLP-encoding. (If it happens to be a byte-string, you can just toss a header in front of it, and then it's RLP-encoded).

In contrast, the docstring does a solid job of explaining: we re-encode typed transactions using RLP. This ensures that transactions are always valid RLP at the outermost layer. Whereas standard transaction encoding is explicitly not valid RLP for typed transactions. So maybe this struct could be named with this approach. Here are some names that might inspire you:

- `TransactionAlwaysRLP`
- `DoublyRLPEncodedTypedTransaction`
- `EnsureOuterRLPTransaction`
- `TransactionForcedToRLPEncoding`

Reply:
While I also don't love my name, I like it more than alternatives that you provided.

I would also add that even with RLP header, encoding is weird and non-usual (compared to standard RLP encoding):
- the type of RLP header is different between legacy and non-legacy tx
  - legacy tx has "list header"
  - non-legacy tx has "byte string header"
- according to [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718), non-legacy transaction doesn't have to be rlp encoded at all (see "Opaque byte array rather than an RLP array" on the linked page)
  - currently they, but EIP leaves space for that to change in the future